src/plugins/formulas/formulas.js

```diff
@@ -121,6 +135,39 @@ export class Formulas extends BasePlugin {
       sheet: this.hyperformula.getSheetId(this.sheetName)
     }
 
+    // TODO if there's an error, somehow retrieve original source value
     valueHolder.value = this.hyperformula.getCellSerialized(address)
   }
+
+  onBeforeCreateRow(row, amount) {
+    return this.hyperformula.isItPossibleToAddRows(this.hyperformula.getSheetId(this.sheetName), [row, amount]);
```

Comment:
```suggestion
    if (this.hyperformula.isItPossibleToAddRows(this.hyperformula.getSheetId(this.sheetName), [row, amount])) {
      return false;
    }
```

... and so on for all `before*` hooks.

The reason is that the next `beforeCreateRow` hook you will register (I've seen TODO to test them) will get as the first argument what was returned by the preceding `beforeCreateRow` hooks. If you return `true`, next hook will get `row = true` instead the correct value.

Reply:
thanks, https://github.com/handsontable/handsontable/pull/7648/commits/5c3d040fffd54ce546f4afb5de7eda8649ab66f0